src/py-opentimelineio/opentimelineio/adapters/svg.py

```diff
@@ -0,0 +1,1011 @@
+
+
+def get_random_color():
+    return [random(), random(), random()]
+
+
+def color_distance(c1, c2):
+    return sum([abs(x[0] - x[1]) for x in zip(c1, c2)])
+
+
+def generate_new_color(existing_colors):
+    max_distance = None
+    best_color = None
+    for i in range(0, 100):
+        color = get_random_color()
+        if not existing_colors:
+            return color
+        best_distance = min([color_distance(color, c) for c in existing_colors])
+        if not max_distance or best_distance > max_distance:
+            max_distance = best_distance
+            best_color = color
+    return best_color
+
+
+def random_color():
+    color = generate_new_color(random_colors_used)
+    random_colors_used.append(color)
+    return color[0] * 255.0, color[1] * 255.0, color[2] * 255.0, 255.0
+
+
+Point = namedtuple('Point', ['x', 'y'])
+
+
+class Rect(object):
+    origin = Point(0, 0)
+    width = 0.0
+    height = 0.0
+
+    def __init__(self, origin=Point(0, 0), width=0.0, height=0.0):
+        self.origin = origin
+        self.width = width
+        self.height = height
+
+    def normalized(self):
+        normalized_origin = Point(
+            self.origin.x + (self.width if self.width < 0 else 0),
+            self.origin.y + (self.height if self.height < 0 else 0),
+        )
+        normalized_width = abs(self.width)
+        normalized_height = abs(self.height)
+        return Rect(normalized_origin, normalized_width, normalized_height)
+
+    def min_x(self):
+        return self.normalized().origin.x
+
+    def min_y(self):
+        return self.normalized().origin.y
+
+    def mid_x(self):
+        return self.origin.x + (self.width * 0.5)
+
+    def mid_y(self):
+        return self.origin.y + (self.height * 0.5)
+
+    def max_x(self):
+        norm = self.normalized()
+        return norm.origin.x + norm.width
+
+    def max_y(self):
+        norm = self.normalized()
+        return norm.origin.y + norm.height
+
+    def contract(self, distance):
+        self.origin.x += distance
+        self.origin.y += distance
+        self.width -= 2.0 * distance
+        self.height -= 2.0 * distance
+
+
+LCARS_CHAR_SIZE_ARRAY = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                         0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                         0, 0, 17, 26, 46, 63, 42, 105, 45, 20, 25, 25, 47, 39, 21,
+                         34, 26, 36, 36, 28, 36, 36, 36,
+                         36, 36, 36, 36, 36, 27, 27, 36, 35, 36, 35, 65, 42, 43, 42,
+                         44, 35, 34, 43, 46, 25, 39, 40,
+                         31, 59, 47, 43, 41, 43, 44, 39, 28, 44, 43, 65, 37, 39, 34,
+                         37, 42, 37, 50, 37, 32, 43, 43,
+                         39, 43, 40, 30, 42, 45, 23, 25, 39, 23, 67, 45, 41, 43, 42,
+                         30, 40, 28, 45, 33, 52, 33, 36,
+                         31, 39, 26, 39, 55]
+image_width = 1000.0
+image_height = 660.0
+font_family = "Roboto"
```

Comment:
Roboto isn't available by default on a lot of systems. For instance, when I view one of the sample SVGs in chrome I see this:
![image](https://user-images.githubusercontent.com/1649087/119546237-8e710a80-bd48-11eb-981f-672b19f92386.png)

Should this be using a font stack that includes some of the more universally available fonts (e.x. Helvetica)?

Reply:
I haven't been able to find a way to use system fonts, or local fonts in an SVG. Here, Roboto will be pulled from the Google Fonts API. If you open the svg in a browser it should work fine.

Reply:
Unfortunately, that doesn't seem to be the case. This screenshot is from Chrome on Mac pointed at this URL: https://raw.githubusercontent.com/PixarAnimationStudios/OpenTimelineIO/7010dc09797dc1f4b6ee922344307f99eaea2f0a/tests/sample_data/transition.svg

Reply:
Oh :(
I'll look into this.

Reply:
Helvetica isn't universally available :)

The only right answer is to embed a font. https://developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/SVG_fonts

Embedding also solves the metrics problem noted above, as they can be fetched from the embed information.

Reply:
I think this may not actually be an answer:
>SVG Fonts are currently supported only in Safari and Android Browser.
Internet Explorer hasn't considered implementing this, the functionality has been removed from Chrome 38 (and Opera 25) and Firefox has postponed its implementation indefinitely to concentrate on WOFF. Other tools however like the Adobe SVG Viewer plugin, Batik and parts of Inkscape support SVG font embedding.

Reply:
The Google fonts thing seems to work on chrome on Android (Not sure if it's because Roboto is the default android font 😅 ). But it worked on chrome on Ubuntu, AFAIR (from the screenshots I posted in this PR).

Reply:
omg it's like the 20th century html font dance. If that works, it satisfies "i don't have Helvetica"., and just leaves us needing to avoid embedding our own guestimate font metrics

Reply:
`font-family="Roboto, Helvetica, Arial, sans-serif"` This didn't work. Using just a single font family is what seems to work. So I've used Arial for now.